AccountingSystem/private/Policies/FilterExecutor.py

```diff
@@ -1,41 +1,41 @@
 
-import types
 from DIRAC import S_OK, S_ERROR, gLogger
 
-class FilterExecutor:
+
+class FilterExecutor(object):
 
   ALLKW = "all"
 
-  def __init__( self ):
+  def __init__(self):
     self.__filters = {}
     self.__globalFilters = []
 
-  def applyFilters( self, iD, credDict, condDict, groupingList ):
-    filters2Apply = list( self.__globalFilters )
+  def applyFilters(self, iD, credDict, condDict, groupingList):
+    filters2Apply = list(self.__globalFilters)
     if iD in self.__filters:
-      filters2Apply.extend( self.__filters[ iD ] )
+      filters2Apply.extend(self.__filters[iD])
     for myFilter in filters2Apply:
       try:
-        gLogger.info( "Applying filter %s for %s" % ( myFilter.__name__, iD ) )
-        retVal = myFilter( credDict, condDict, groupingList )
-        if not retVal[ 'OK' ]:
-          gLogger.info( "Filter %s for %s failed: %s" % ( myFilter.__name__, iD, retVal[ 'Message' ] ) )
+        gLogger.info("Applying filter %s for %s" % (myFilter.__name__, iD))
+        retVal = myFilter(credDict, condDict, groupingList)
+        if not retVal['OK']:
+          gLogger.info("Filter %s for %s failed: %s" % (myFilter.__name__, iD, retVal['Message']))
           return retVal
-      except:
-        gLogger.exception( "Exception while applying filter", "%s for %s" % ( myFilter.__name__, iD ) )
-        return S_ERROR( "Exception while applying filters" )
+      except Exception:
+        gLogger.exception("Exception while applying filter", "%s for %s" % (myFilter.__name__, iD))
+        return S_ERROR("Exception while applying filters")
     return S_OK()
 
-  def addFilter( self, iD, myFilter ):
+  def addFilter(self, iD, myFilter):
     if iD not in self.__filters:
-      self.__filters[ iD ] = []
-    if type( myFilter ) in ( types.ListType, types.TupleType ):
-      self.__filters[ iD ].extend( myFilter )
+      self.__filters[iD] = []
+    if type(myFilter) in (list, tuple):
```

Comment:
```suggestion
    if isinstance(myFilter, (list, tuple)):
```
or use `from collections import Iterable` (it's removed in 3.8 but that's a problem for another day)
```suggestion
    if isinstance(myFilter, Iterable):
```